src/DotNet.Testcontainers/Clients/TestcontainersClient.cs

```diff
@@ -192,6 +192,44 @@ await this.containers.ExtractArchiveToContainerAsync(id, "/", memStream, ct)
       }
     }
 
+    /// <inheritdoc />
+    public async Task<byte[]> CopyFileFromContainerAsync(string id, string filePath, CancellationToken ct = default)
+    {
+      Stream archive;
+      try
+      {
+        archive = await this.containers.GetArchiveFromContainerAsync(id, filePath, ct)
+          .ConfigureAwait(false);
+      }
+      catch (DockerContainerNotFoundException ex)
+      {
+        throw new InvalidOperationException(
+          $"Not possible to get archive with file {filePath} from container: {ex.ResponseBody}",
+          ex);
+      }
+
+      using (var tarInputStream = new TarInputStream(archive, Encoding.Default))
+      {
+        tarInputStream.IsStreamOwner = true;
+        var entry = tarInputStream.GetNextEntry();
+        if (entry == null || entry.IsDirectory || entry.Name != Path.GetFileName(filePath))
```

Comment:
I've added some minor improvements. Before I'll merge the pull request, I'd like to ask if this couldn't be simplified to:

```csharp
if (entry.IsDirectory)
{
  throw new InvalidOperationException("Can not read from a directory. Use a file instead.");
}
```

I think we don't need the other conditions. If the file (path) does not exist, `GetArchiveFromContainerAsync` will already fail.

Reply:
Yes, that is correct.